opentelemetry-otlp/src/exporter/http/mod.rs

```diff
@@ -348,9 +346,7 @@ impl OtlpHttpClient {
             #[cfg(feature = "http-json")]
             Protocol::HttpJson => match serde_json::to_string_pretty(&req) {
                 Ok(json) => Ok((json.into(), "application/json")),
-                Err(e) => Err(opentelemetry_sdk::metrics::MetricError::Other(
-                    e.to_string(),
-                )),
+                Err(e) => Err(Box::new(e)),
```

Comment:
Same as above - should we return the serde error when it's the only failure that can shake out of this?